src/lib/components/Table.svelte

```diff
@@ -174,56 +175,60 @@
                     {/each}
                 </tr>
             </thead>
-            <tbody>
-                {#if isLoading}
-                    <tr>
-                        <td colspan="99">
-                            <CenteredSpinner />
-                        </td>
-                    </tr>
-                {:else}
-                    {#each items as item, index (idColumn === 'index' ? index : item[idColumn])}
-                        {@const href =
-                            itemUrlPrefix && `${itemUrlPrefix}${idColumn === 'index' ? '' : item[idColumn] ?? ''}`}
-                        <tr class="hover">
-                            {#if enableSelectAll || enableSelect}
-                                <TableCell class="w-4">
-                                    <input
-                                        type="checkbox"
-                                        class="checkbox checkbox-sm"
-                                        on:change={() => onSelectItem(item)}
-                                        checked={selectedItems.includes(item)}
-                                    />
-                                </TableCell>
-                            {/if}
-                            {#each columns as { itemKey, text, width } (itemKey)}
-                                <slot {item} {href} {itemKey} columnText={text}>
-                                    {#if href !== undefined && itemKey}
-                                        <LinkedTableCell style={width ? `width: ${width}ch;` : ''} {href}
-                                            >{item[itemKey]?.toLocaleString() ?? ''}</LinkedTableCell
-                                        >
-                                    {:else if itemKey !== undefined}
-                                        <TableCell style={width ? `width: ${width}ch;` : ''}
-                                            >{item[itemKey]?.toLocaleString() ?? ''}</TableCell
-                                        >
-                                    {/if}
-                                </slot>
-                            {/each}
-                        </tr>
-                    {/each}
-                    {#if items.length === 0}
+            {#if customTbody}
+                <slot name="customTbody" rowItems={items} {columns} />
+            {:else}
```

Comment:
I think we could avoid needing the extra prop `customTbody` prop if we use the <slot> and pass the <tbody> into it. Since Svelte will use whatever's inside the slot component as the fallback:
```
<slot name="customTbody" rowItems={items} {columns}>
  <tbody>
    {#if isLoading}
    ....
  </tbody>
</slot>
```